packages/app/src/messages/reducers/messages.interface.ts

```diff
@@ -87,11 +87,20 @@ export type AddNewMessageAction = PayloadAction<
   MessagesEvent.AddNewMessage
 >
 
-export type ToggleThreadReadStatusAction = PayloadAction<
-  string[],
-  MessagesEvent.ToggleThreadReadStatus
+export type ToggleThreadReadStatusPendingAction = PayloadAction<
+  undefined,
+  MessagesEvent.ToggleThreadReadStatus,
+  {
+    arg: Thread[]
+  }
+>
+export type MarkThreadsReadStatusPendingAction = PayloadAction<
+  undefined,
+  MessagesEvent.ToggleThreadReadStatus,
+  {
+    arg: Thread[]
+  }
```

Comment:
Maybe just `threads` instead of `arg`?

Reply:
Good intuition. However `arg` is a part of the API.